pkg/commands/os.go

```diff
@@ -57,6 +58,53 @@ func (c *OSCommand) RunCommandWithOutput(command string) (string, error) {
 	)
 }
 
+// RunCommandWithOutputLive runs RunCommandWithOutputLiveWrapper
+func (c *OSCommand) RunCommandWithOutputLive(command string, output func(string) string) (errorMessage string, err error) {
+	return RunCommandWithOutputLiveWrapper(c, command, output)
+}
+
+// DetectUnamePass detect a username / password question in a command
+// ask is a function that gets executen when this function detect you need to fillin a password
+// The ask argument will be "username" or "password" and expects the user's password or username back
+func (c *OSCommand) DetectUnamePass(command string, ask func(string) string) error {
+	ttyText := ""
+	errMessage, err := c.RunCommandWithOutputLive(command, func(word string) string {
+		ttyText = ttyText + " " + word
+
+		type Prompt struct {
+			pattern   string
+			canAskFor bool
+		}
+		prompts := map[string]Prompt{
+			"password": {
+				pattern:   `Password\s*for\s*'.+':`,
+				canAskFor: true,
+			},
+			"username": {
+				pattern:   `Username\s*for\s*'.+':`,
+				canAskFor: true,
+			},
+		}
+
+		for askFor, prompt := range prompts {
+			if match, _ := regexp.MatchString(prompt.pattern, ttyText); match && prompt.canAskFor {
+				prompt.canAskFor = false
+				ttyText = ""
+				return ask(askFor)
+			}
+		}
+
+		return ""
+	})
+	if err != nil {
+		if strings.Contains("exit status 128", err.Error()) {
```

Comment:
given that this error ends up being re-checked below, maybe we should just return it as it is here and then do the checking from the caller?